.github/workflows/test-integration.yml

```diff
@@ -0,0 +1,321 @@
+name: Integration Tests
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
```

Comment:
do you think on: push is necessary?

Since we only have concurrency 1 for the integration tests, this might unnecessarily block resources

Reply:
Yeah we can skip on push with the merge queue.